src/Simplify.elm

```diff
@@ -3875,6 +3882,65 @@ listAppendChecks checkInfo =
             []
 
 
+listHeadChecks : CheckInfo -> List (Error {})
+listHeadChecks checkInfo =
+    let
+        justFirstElementError : Range -> List (Error {})
+        justFirstElementError keep =
+            [ Rule.errorWithFix
+                { message = "Using List.head on a list with a first element will result in Just that element"
+                , details = [ "You can replace this call by Just the first list element." ]
+                }
+                checkInfo.fnRange
+                (keepOnlyFix { parentRange = Node.range listArg, keep = keep }
```

Comment:
I'm thinking we may need to add parens around `keep`. There is `needsParens` to help determine that.
```elm
List.head [ f a ]
--> Just f a ❌
--> Just (f a) ✅
```

Reply:
`needsParens` is :ok_hand: